stdlib/public/TensorFlow/TensorGroup.swift

```diff
@@ -0,0 +1,43 @@
+//===-- TensorGroup.swift ---------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the TensorGroup type.
+//
+//===----------------------------------------------------------------------===//
+
+/// A protocol for types that can be used as tensor operation inputs and
+/// outputs. When a TensorGroup is used as an input, it gets passed to the
+/// tensor operation as an input list whose elements are the tensor fields of
+/// the type. When a TensorGroup is used as an output, it gets initialized
+/// with its tensor fields set to the tensor operation's tensor outputs.
+///
+/// TODO: Add a derived conformance to TensorGroup so that users don't have
+/// to write the conformance themselves.
+public protocol TensorGroup {
+  /// The number of tensor fields in this type.
+  static var _tensorHandleCount: Int32 { get }
```

Comment:
why do we need this property, instead of having caller just use `_unknownShapeList.count`?

Reply:
can we remove this stored property, and instead use `_typeList.count`? 

(we can make this a computed property if that makes caller's code simpler somehow.)

Reply:
done